app/scripts/modules/amazon/src/loadBalancer/configure/network/NLBListeners.tsx

```diff
@@ -68,28 +147,14 @@ export class NLBListeners
     return errors;
   }
 
-  private needsCert(listener: IListenerDescription): boolean {
-    return listener.protocol === 'TLS';
-  }
-
-  private showCertificateSelect(certificate: IALBListenerCertificate): boolean {
-    return certificate.type === 'iam' && this.state.certificates && Object.keys(this.state.certificates).length > 0;
+  private updateListeners(): void {
+    this.props.formik.setFieldValue('listeners', this.props.formik.values.listeners);
```

Comment:
I think we can remove this in favor of `updateListeners` in the `CertificateSelectorComponent`.

Reply:
Discussed with @caseyhebebrand offline and resolved this question , the last commit has all the changes.